source/rust_verify/tests/strings.rs

```diff
@@ -0,0 +1,87 @@
+#![feature(rustc_private)]
```

Comment:
Please add tests for:
* inequality and equality of string literals, both as they appear inline and as part of spec functions or e.g. postconditions of functions, i.e.
```
fn hello_str<'a>() -> (res: StrSlice<'a>)
  ensures res.substring(0, 5) === new_strlit("Hello")
```
and a function that uses the result of `hello_str`.

Reply:
Also, please add tests with multiple StrSlices and literals in the same program.